src/swarm-lang/Swarm/Language/Typecheck.hs

```diff
@@ -127,6 +127,19 @@ type Sourced a = (Source, a)
 
 -- | A "join" where an expected thing meets an actual thing.
 newtype Join a = Join (Source -> a)
+  deriving (Functor)
+
+instance Foldable Join where
+  foldMap :: Monoid m => (a -> m) -> Join a -> m
```

Comment:
Is this used anywhere?

Reply:
Not directly, but `Foldable` is a prerequisite of `Traversable`.